src/Ghcid.hs

```diff
@@ -156,7 +159,18 @@ autoOptions o@Options{..}
                                 "stack exec --test --bench -- ghci" : opts
                 in f flags $ stack:cabal
               | ".ghci" `elem` files -> f ("ghci":opts) [curdir </> ".ghci"]
-              | cabal /= [] -> f (if null arguments then "cabal repl":map ("--ghc-options=" ++) opts else "cabal exec -- ghci":opts) cabal
+              | cabal /= [] ->
+                  let useCabal =
+                          let prefix =
+                                  [ "cabal", "repl" ] ++ Foldable.toList target
+
+                              suffix = map ("--ghc-options=" ++) opts
+
+                          in  prefix ++ suffix
+
+                      useGHCI = "cabal exec -- ghci":opts
+
+                  in  f (if null arguments then useCabal else useGHCI) cabal
```

Comment:
This section of the code now has a lot of vertical space. The blank lines between each line aren't needed. I do agree it was light on the whitespace before though!